pkg/tsdb/prometheus/querydata/response.go

```diff
@@ -25,12 +25,15 @@ func (s *QueryData) parseResponse(ctx context.Context, q *models.Query, res *htt
 		}
 	}()
 
+	cfg := backend.GrafanaConfigFromContext(ctx)
+	enableDataplane := cfg.FeatureToggles().IsEnabled("prometheusDataplane")
```

Comment:
Nit: Similar to the other comment above, maybe it's worth adding a `bool enableDataplane` argument to this function to avoid parsing the feature flags for each response